packages/app/src/app/overmind/namespaces/workspace/actions.ts

```diff
@@ -444,3 +446,53 @@ export const openDefaultItem: Action = ({ state }) => {
 
   state.workspace.openedWorkspaceItem = defaultItem.id;
 };
+
+export const getDependencies: AsyncAction<string | void> = async (
+  { state, effects },
+  value
+) => {
+  state.workspace.loadingDependencySearch = true;
+  const searchResults = await effects.algoliaSearch.searchDependencies(value);
+
+  state.workspace.loadingDependencySearch = false;
+
+  if (!value) {
+    state.workspace.starterDependencies = [...searchResults];
+  }
+
+  state.workspace.dependencies = [...searchResults];
+};
+
+export const setDependencies: Action<Dependency[]> = (
+  { state },
+  newDependencies
+) => {
+  state.workspace.dependencies = [...newDependencies];
+};
+
+export const setSelectedDependencies: Action<Dependency> = (
+  { state },
+  dependency
+) => {
+  const oldDeps = state.workspace.selectedDependencies;
+  state.workspace.selectedDependencies = {
+    ...oldDeps,
+    [dependency.objectID]: oldDeps[dependency.objectID] ? null : dependency,
+  };
+};
+
+export const handleVersionChange: Action<{
+  dependency: Dependency;
+  version: string;
+}> = ({ state }, { dependency, version }) => {
+  json(state.workspace.hitToVersionMap).set(dependency.objectID, version);
```

Comment:
Does `hitToVersionMap` need to be a map? Could just be a plain object? Maps are really only necessary when the key is something else than a string/number 😄

Reply:
in this case it is, its the whole dependency, it does need to be one :/

Reply:
Oh, `objectID` is the whole dependency? Does it not have some string representing its id? I thought `objectID` was like the name of the dependency?